<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_76) on Wed Apr 06 15:06:58 UTC 2016 -->
<title>GenomicRegionSearchService (intermine-javadoc)</title>
<meta name="date" content="2016-04-06">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="GenomicRegionSearchService (intermine-javadoc)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchQueryRunner.html" title="class in org.intermine.bio.web.logic"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchUtil.html" title="class in org.intermine.bio.web.logic"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/intermine/bio/web/logic/GenomicRegionSearchService.html" target="_top">Frames</a></li>
<li><a href="GenomicRegionSearchService.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.intermine.bio.web.logic</div>
<h2 title="Class GenomicRegionSearchService" class="title">Class GenomicRegionSearchService</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>org.intermine.bio.web.logic.GenomicRegionSearchService</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">GenomicRegionSearchService</span>
extends java.lang.Object</pre>
<div class="block">A class to provide genomic region search services in general.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#DEFAULT_REGION_INIT_BATCH_SIZE">DEFAULT_REGION_INIT_BATCH_SIZE</a></strong></code>
<div class="block">Default batch size to be used for region search initialisation queries.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#GenomicRegionSearchService()">GenomicRegionSearchService</a></strong>()</code>
<div class="block">Constructor</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#categorizeFeatureTypes(java.util.Set,%20org.intermine.bio.web.model.GenomicRegion)">categorizeFeatureTypes</a></strong>(java.util.Set&lt;java.lang.String&gt;&nbsp;featureSet,
                      <a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;s)</code>
<div class="block">Get all feature types from a list of sequence features</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#convertResultMapToHTML(java.util.Map,%20java.util.Map,%20java.util.List,%20int,%20int,%20javax.servlet.http.HttpSession)">convertResultMapToHTML</a></strong>(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap,
                      java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&gt;&nbsp;resultStat,
                      java.util.List&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&gt;&nbsp;genomicRegionList,
                      int&nbsp;fromIdx,
                      int&nbsp;toIdx,
                      javax.servlet.http.HttpSession&nbsp;session)</code>
<div class="block">Convert result map to HTML string.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,<a href="../../../../../org/intermine/objectstore/query/Query.html" title="class in org.intermine.objectstore.query">Query</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#createQueryList()">createQueryList</a></strong>()</code>
<div class="block">To prepare queries for genomic regions</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#generateCreateListHtml(java.util.Map)">generateCreateListHtml</a></strong>(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)</code>
<div class="block">Generate a html string with all feature type for list creation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#generateJBrowseURL(org.intermine.bio.web.model.GenomicRegion)">generateJBrowseURL</a></strong>(<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;s)</code>
<div class="block">A wrapper of generateJBrowseURL(GenomicRegion s, List<String> featureTypes)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#generateJBrowseURL(org.intermine.bio.web.model.GenomicRegion,%20java.util.List)">generateJBrowseURL</a></strong>(<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;s,
                  java.util.List&lt;java.lang.String&gt;&nbsp;featureTypes)</code>
<div class="block">Generated a jbrowse url</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getAllGenomicRegionOverlapFeaturesByType(java.util.Map,%20java.lang.String)">getAllGenomicRegionOverlapFeaturesByType</a></strong>(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap,
                                        java.lang.String&nbsp;featureType)</code>
<div class="block">Get a set of ids of all span's overlap features by given feature type. for
 GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,java.util.Map&lt;java.lang.String,<a href="../../../../../org/intermine/bio/web/model/ChromosomeInfo.html" title="class in org.intermine.bio.web.model">ChromosomeInfo</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getChromosomeInfomationMap()">getChromosomeInfomationMap</a></strong>()</code>
<div class="block">Get chromosome information as in a map, keys are lowercased chromosome ids</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/intermine/bio/web/model/GenomicRegionSearchConstraint.html" title="class in org.intermine.bio.web.model">GenomicRegionSearchConstraint</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getConstraint()">getConstraint</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/intermine/pathquery/PathQuery.html" title="class in org.intermine.pathquery">PathQuery</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getExportFeaturesQuery(java.util.Set,%20java.lang.String,%20java.util.Set,%20java.util.List)">getExportFeaturesQuery</a></strong>(java.util.Set&lt;java.lang.Integer&gt;&nbsp;featureIds,
                      java.lang.String&nbsp;featureType,
                      java.util.Set&lt;java.lang.String&gt;&nbsp;views,
                      java.util.List&lt;java.lang.String&gt;&nbsp;sortOrder)</code>
<div class="block">A flexiable way of setting query fields.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getFeatureTypeSetInAlphabeticalOrder(java.util.Set)">getFeatureTypeSetInAlphabeticalOrder</a></strong>(java.util.Set&lt;java.lang.String&gt;&nbsp;featureSet)</code>
<div class="block">Get all feature types in a  TresSet</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,java.util.List&lt;java.lang.String&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getFeatureTypeToSOTermMap()">getFeatureTypeToSOTermMap</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getGenomicRegionExtendedSizeConstraint(java.lang.String,%20java.util.Map)">getGenomicRegionExtendedSizeConstraint</a></strong>(java.lang.String&nbsp;spanUUIDString,
                                      java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegionSearchConstraint.html" title="class in org.intermine.bio.web.model">GenomicRegionSearchConstraint</a>,java.lang.String&gt;&nbsp;spanConstraintMap)</code>
<div class="block">Get flanking size for GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getGenomicRegionOrganismConstraint(java.lang.String,%20java.util.Map)">getGenomicRegionOrganismConstraint</a></strong>(java.lang.String&nbsp;spanUUIDString,
                                  java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegionSearchConstraint.html" title="class in org.intermine.bio.web.model">GenomicRegionSearchConstraint</a>,java.lang.String&gt;&nbsp;spanConstraintMap)</code>
<div class="block">Get organism for GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getGenomicRegionOverlapFeaturesAsSet(java.lang.String,%20java.util.Map)">getGenomicRegionOverlapFeaturesAsSet</a></strong>(java.lang.String&nbsp;grInfo,
                                    java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)</code>
<div class="block">Get a set of ids of a span's overlap features. for
 GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getGenomicRegionOverlapFeaturesAsString(java.lang.String,%20java.util.Map)">getGenomicRegionOverlapFeaturesAsString</a></strong>(java.lang.String&nbsp;grInfo,
                                       java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)</code>
<div class="block">Get a comma separated string of a span's overlap features. for
 GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getGenomicRegionOverlapFeaturesByType(java.lang.String,%20java.util.Map,%20java.lang.String)">getGenomicRegionOverlapFeaturesByType</a></strong>(java.lang.String&nbsp;grInfo,
                                     java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap,
                                     java.lang.String&nbsp;featureType)</code>
<div class="block">Get a set of ids of a span's overlap features by given feature type. for
 GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,java.lang.String&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getJBrowseTracks()">getJBrowseTracks</a></strong>()</code>
<div class="block">Get jbrowse track information from web.properties</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getMatchedBaseCount(org.intermine.bio.web.model.GenomicRegion,%20java.util.List)">getMatchedBaseCount</a></strong>(<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;gr,
                   java.util.List&lt;java.lang.String&gt;&nbsp;r)</code>
<div class="block">Calculate the number of matched bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getOptionsCss()">getOptionsCss</a></strong>()</code>
<div class="block">Get the name of customized options CSS</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getOptionsJavascript()">getOptionsJavascript</a></strong>()</code>
<div class="block">Get the name of customized options javascript, by default, it is named
 "genomic_region_search_options_default.js"</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,java.lang.Integer&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getOrganismToTaxonMap()">getOrganismToTaxonMap</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getResultsCss()">getResultsCss</a></strong>()</code>
<div class="block">Get the name of customized results CSS</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getResultsJavascript()">getResultsJavascript</a></strong>()</code>
<div class="block">Get the name of customized results javascript</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.util.List&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getSelectionInformation()">getSelectionInformation</a></strong>()</code>
<div class="block">The message passed to results page</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Set&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#getTaxonIds(java.util.Set)">getTaxonIds</a></strong>(java.util.Set&lt;java.lang.String&gt;&nbsp;organisms)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#hasJBrowseTrack(java.lang.String)">hasJBrowseTrack</a></strong>(java.lang.String&nbsp;featureType)</code>
<div class="block">Test if jbrowse has a track regards to a feature type</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#init(javax.servlet.http.HttpServletRequest)">init</a></strong>(javax.servlet.http.HttpServletRequest&nbsp;request)</code>
<div class="block">To set globally used variables.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#isEmptyFeature(java.util.Map)">isEmptyFeature</a></strong>(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)</code>
<div class="block">Check whether the results have empty features. for
 GenomicRegionSearchAjaxAction use.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#isJBrowseEnabled()">isJBrowseEnabled</a></strong>()</code>
<div class="block">Test if jbrowse is enabled</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>org.apache.struts.action.ActionMessage</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#parseBasicInput(org.intermine.bio.web.struts.GenomicRegionSearchForm)">parseBasicInput</a></strong>(<a href="../../../../../org/intermine/bio/web/struts/GenomicRegionSearchForm.html" title="class in org.intermine.bio.web.struts">GenomicRegionSearchForm</a>&nbsp;grsForm)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>org.apache.struts.action.ActionMessage</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#parseGenomicRegionSearchForm(org.intermine.bio.web.struts.GenomicRegionSearchForm)">parseGenomicRegionSearchForm</a></strong>(<a href="../../../../../org/intermine/bio/web/struts/GenomicRegionSearchForm.html" title="class in org.intermine.bio.web.struts">GenomicRegionSearchForm</a>&nbsp;grsForm)</code>
<div class="block">To parse form data</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#setupWebData()">setupWebData</a></strong>()</code>
<div class="block">To call the queryOrganismAndSequenceFeatureTypes method in
 GenomicRegionSearchQueryRunner.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>java.util.Map&lt;java.lang.String,java.util.List&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchService.html#validateGenomicRegions()">validateGenomicRegions</a></strong>()</code>
<div class="block">Validate input genomic regions</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="DEFAULT_REGION_INIT_BATCH_SIZE">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>DEFAULT_REGION_INIT_BATCH_SIZE</h4>
<pre>public static final&nbsp;int DEFAULT_REGION_INIT_BATCH_SIZE</pre>
<div class="block">Default batch size to be used for region search initialisation queries.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#org.intermine.bio.web.logic.GenomicRegionSearchService.DEFAULT_REGION_INIT_BATCH_SIZE">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GenomicRegionSearchService()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenomicRegionSearchService</h4>
<pre>public&nbsp;GenomicRegionSearchService()</pre>
<div class="block">Constructor</div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="init(javax.servlet.http.HttpServletRequest)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>init</h4>
<pre>public&nbsp;void&nbsp;init(javax.servlet.http.HttpServletRequest&nbsp;request)</pre>
<div class="block">To set globally used variables.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>request</code> - HttpServletRequest</dd></dl>
</li>
</ul>
<a name="setupWebData()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setupWebData</h4>
<pre>public&nbsp;java.lang.String&nbsp;setupWebData()
                              throws java.lang.Exception</pre>
<div class="block">To call the queryOrganismAndSequenceFeatureTypes method in
 GenomicRegionSearchQueryRunner.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a JSON string</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - e</dd></dl>
</li>
</ul>
<a name="getOptionsJavascript()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOptionsJavascript</h4>
<pre>public&nbsp;java.lang.String&nbsp;getOptionsJavascript()</pre>
<div class="block">Get the name of customized options javascript, by default, it is named
 "genomic_region_search_options_default.js"</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the name of options javascript name</dd></dl>
</li>
</ul>
<a name="getResultsJavascript()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getResultsJavascript</h4>
<pre>public&nbsp;java.lang.String&nbsp;getResultsJavascript()</pre>
<div class="block">Get the name of customized results javascript</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the name of results page</dd></dl>
</li>
</ul>
<a name="getOptionsCss()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOptionsCss</h4>
<pre>public&nbsp;java.lang.String&nbsp;getOptionsCss()</pre>
<div class="block">Get the name of customized options CSS</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the name of options css</dd></dl>
</li>
</ul>
<a name="getResultsCss()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getResultsCss</h4>
<pre>public&nbsp;java.lang.String&nbsp;getResultsCss()</pre>
<div class="block">Get the name of customized results CSS</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the name of results css</dd></dl>
</li>
</ul>
<a name="parseGenomicRegionSearchForm(org.intermine.bio.web.struts.GenomicRegionSearchForm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>parseGenomicRegionSearchForm</h4>
<pre>public&nbsp;org.apache.struts.action.ActionMessage&nbsp;parseGenomicRegionSearchForm(<a href="../../../../../org/intermine/bio/web/struts/GenomicRegionSearchForm.html" title="class in org.intermine.bio.web.struts">GenomicRegionSearchForm</a>&nbsp;grsForm)
                                                                    throws java.lang.Exception</pre>
<div class="block">To parse form data</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>grsForm</code> - GenomicRegionSearchForm</dd>
<dt><span class="strong">Returns:</span></dt><dd>genomic region search constraint</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - e</dd></dl>
</li>
</ul>
<a name="parseBasicInput(org.intermine.bio.web.struts.GenomicRegionSearchForm)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>parseBasicInput</h4>
<pre>public&nbsp;org.apache.struts.action.ActionMessage&nbsp;parseBasicInput(<a href="../../../../../org/intermine/bio/web/struts/GenomicRegionSearchForm.html" title="class in org.intermine.bio.web.struts">GenomicRegionSearchForm</a>&nbsp;grsForm)
                                                       throws java.lang.Exception</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>grsForm</code> - GenomicRegionSearchForm</dd>
<dt><span class="strong">Returns:</span></dt><dd>ActionMessage</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - e</dd></dl>
</li>
</ul>
<a name="createQueryList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createQueryList</h4>
<pre>public&nbsp;java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,<a href="../../../../../org/intermine/objectstore/query/Query.html" title="class in org.intermine.objectstore.query">Query</a>&gt;&nbsp;createQueryList()</pre>
<div class="block">To prepare queries for genomic regions</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a list of prepared queries for genomic regions</dd></dl>
</li>
</ul>
<a name="getConstraint()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getConstraint</h4>
<pre>public&nbsp;<a href="../../../../../org/intermine/bio/web/model/GenomicRegionSearchConstraint.html" title="class in org.intermine.bio.web.model">GenomicRegionSearchConstraint</a>&nbsp;getConstraint()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the grsc</dd></dl>
</li>
</ul>
<a name="getChromosomeInfomationMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeInfomationMap</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,java.util.Map&lt;java.lang.String,<a href="../../../../../org/intermine/bio/web/model/ChromosomeInfo.html" title="class in org.intermine.bio.web.model">ChromosomeInfo</a>&gt;&gt;&nbsp;getChromosomeInfomationMap()</pre>
<div class="block">Get chromosome information as in a map, keys are lowercased chromosome ids</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>chrInfoMap</dd></dl>
</li>
</ul>
<a name="getFeatureTypeToSOTermMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatureTypeToSOTermMap</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,java.util.List&lt;java.lang.String&gt;&gt;&nbsp;getFeatureTypeToSOTermMap()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>featureTypeToSOTermMap</dd></dl>
</li>
</ul>
<a name="getOrganismToTaxonMap()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOrganismToTaxonMap</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&nbsp;getOrganismToTaxonMap()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>orgTaxonIdMap</dd></dl>
</li>
</ul>
<a name="validateGenomicRegions()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>validateGenomicRegions</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,java.util.List&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&gt;&gt;&nbsp;validateGenomicRegions()
                                                                                     throws java.lang.Exception</pre>
<div class="block">Validate input genomic regions</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>resultMap</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - with error message</dd></dl>
</li>
</ul>
<a name="getSelectionInformation()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSelectionInformation</h4>
<pre>public&nbsp;java.util.List&lt;java.lang.String&gt;&nbsp;getSelectionInformation()</pre>
<div class="block">The message passed to results page</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>resultMessages</dd></dl>
</li>
</ul>
<a name="getGenomicRegionOrganismConstraint(java.lang.String, java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomicRegionOrganismConstraint</h4>
<pre>public&nbsp;java.lang.String&nbsp;getGenomicRegionOrganismConstraint(java.lang.String&nbsp;spanUUIDString,
                                                  java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegionSearchConstraint.html" title="class in org.intermine.bio.web.model">GenomicRegionSearchConstraint</a>,java.lang.String&gt;&nbsp;spanConstraintMap)</pre>
<div class="block">Get organism for GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>spanUUIDString</code> - uuid</dd><dd><code>spanConstraintMap</code> - map of contraints</dd>
<dt><span class="strong">Returns:</span></dt><dd>the organism</dd></dl>
</li>
</ul>
<a name="getGenomicRegionExtendedSizeConstraint(java.lang.String, java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomicRegionExtendedSizeConstraint</h4>
<pre>public&nbsp;int&nbsp;getGenomicRegionExtendedSizeConstraint(java.lang.String&nbsp;spanUUIDString,
                                         java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegionSearchConstraint.html" title="class in org.intermine.bio.web.model">GenomicRegionSearchConstraint</a>,java.lang.String&gt;&nbsp;spanConstraintMap)</pre>
<div class="block">Get flanking size for GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>spanUUIDString</code> - uuid</dd><dd><code>spanConstraintMap</code> - map of contraints</dd>
<dt><span class="strong">Returns:</span></dt><dd>the flanking size</dd></dl>
</li>
</ul>
<a name="getGenomicRegionOverlapFeaturesAsSet(java.lang.String, java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomicRegionOverlapFeaturesAsSet</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.Integer&gt;&nbsp;getGenomicRegionOverlapFeaturesAsSet(java.lang.String&nbsp;grInfo,
                                                                    java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)
                                                                      throws java.lang.Exception</pre>
<div class="block">Get a set of ids of a span's overlap features. for
 GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>grInfo</code> - a genomic region in string array</dd><dd><code>resultMap</code> - map of search results</dd>
<dt><span class="strong">Returns:</span></dt><dd>String feature ids joined by comma</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - with error message</dd></dl>
</li>
</ul>
<a name="getGenomicRegionOverlapFeaturesByType(java.lang.String, java.util.Map, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomicRegionOverlapFeaturesByType</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.Integer&gt;&nbsp;getGenomicRegionOverlapFeaturesByType(java.lang.String&nbsp;grInfo,
                                                                     java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap,
                                                                     java.lang.String&nbsp;featureType)
                                                                       throws java.lang.Exception</pre>
<div class="block">Get a set of ids of a span's overlap features by given feature type. for
 GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>grInfo</code> - a genomic region in string array</dd><dd><code>resultMap</code> - map of search results</dd><dd><code>featureType</code> - e.g. Gene</dd>
<dt><span class="strong">Returns:</span></dt><dd>String feature ids joined by comma</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - with error message</dd></dl>
</li>
</ul>
<a name="getAllGenomicRegionOverlapFeaturesByType(java.util.Map, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllGenomicRegionOverlapFeaturesByType</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.Integer&gt;&nbsp;getAllGenomicRegionOverlapFeaturesByType(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap,
                                                                        java.lang.String&nbsp;featureType)</pre>
<div class="block">Get a set of ids of all span's overlap features by given feature type. for
 GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>resultMap</code> - map of search results</dd><dd><code>featureType</code> - e.g. Gene</dd>
<dt><span class="strong">Returns:</span></dt><dd>String feature ids joined by comma</dd></dl>
</li>
</ul>
<a name="getGenomicRegionOverlapFeaturesAsString(java.lang.String, java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGenomicRegionOverlapFeaturesAsString</h4>
<pre>public&nbsp;java.lang.String&nbsp;getGenomicRegionOverlapFeaturesAsString(java.lang.String&nbsp;grInfo,
                                                       java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)
                                                         throws java.lang.Exception</pre>
<div class="block">Get a comma separated string of a span's overlap features. for
 GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>grInfo</code> - a genomic region in string array</dd><dd><code>resultMap</code> - map of search results</dd>
<dt><span class="strong">Returns:</span></dt><dd>String feature ids joined by comma</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.lang.Exception</code> - with error message</dd></dl>
</li>
</ul>
<a name="isEmptyFeature(java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isEmptyFeature</h4>
<pre>public&nbsp;java.lang.String&nbsp;isEmptyFeature(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)</pre>
<div class="block">Check whether the results have empty features. for
 GenomicRegionSearchAjaxAction use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>resultMap</code> - map of search results</dd>
<dt><span class="strong">Returns:</span></dt><dd>String "hasFeature" or "emptyFeature"</dd></dl>
</li>
</ul>
<a name="generateCreateListHtml(java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>generateCreateListHtml</h4>
<pre>public&nbsp;java.lang.String&nbsp;generateCreateListHtml(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap)</pre>
<div class="block">Generate a html string with all feature type for list creation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>resultMap</code> - map of search results</dd>
<dt><span class="strong">Returns:</span></dt><dd>a html string</dd></dl>
</li>
</ul>
<a name="convertResultMapToHTML(java.util.Map, java.util.Map, java.util.List, int, int, javax.servlet.http.HttpSession)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertResultMapToHTML</h4>
<pre>public&nbsp;java.lang.String&nbsp;convertResultMapToHTML(java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.List&lt;java.util.List&lt;java.lang.String&gt;&gt;&gt;&nbsp;resultMap,
                                      java.util.Map&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>,java.util.Map&lt;java.lang.String,java.lang.Integer&gt;&gt;&nbsp;resultStat,
                                      java.util.List&lt;<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&gt;&nbsp;genomicRegionList,
                                      int&nbsp;fromIdx,
                                      int&nbsp;toIdx,
                                      javax.servlet.http.HttpSession&nbsp;session)</pre>
<div class="block">Convert result map to HTML string.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>resultMap</code> - resultMap</dd><dd><code>resultStat</code> - result statistics</dd><dd><code>genomicRegionList</code> - spanList</dd><dd><code>fromIdx</code> - offsetStart</dd><dd><code>toIdx</code> - offsetEnd</dd><dd><code>session</code> - the current session</dd>
<dt><span class="strong">Returns:</span></dt><dd>a String of HTML</dd></dl>
</li>
</ul>
<a name="categorizeFeatureTypes(java.util.Set, org.intermine.bio.web.model.GenomicRegion)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>categorizeFeatureTypes</h4>
<pre>public&nbsp;java.lang.String&nbsp;categorizeFeatureTypes(java.util.Set&lt;java.lang.String&gt;&nbsp;featureSet,
                                      <a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;s)</pre>
<div class="block">Get all feature types from a list of sequence features</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featureSet</code> - a set of feature types (class names) in a special order</dd><dd><code>s</code> - GenomicRegion</dd>
<dt><span class="strong">Returns:</span></dt><dd>A html string with a dropdown list of feature types</dd></dl>
</li>
</ul>
<a name="getFeatureTypeSetInAlphabeticalOrder(java.util.Set)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatureTypeSetInAlphabeticalOrder</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.String&gt;&nbsp;getFeatureTypeSetInAlphabeticalOrder(java.util.Set&lt;java.lang.String&gt;&nbsp;featureSet)</pre>
<div class="block">Get all feature types in a  TresSet</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featureSet</code> - a set of feature types (class names) in a special order</dd>
<dt><span class="strong">Returns:</span></dt><dd>a set of feature types of a genomic region in alphabetical order</dd></dl>
</li>
</ul>
<a name="getMatchedBaseCount(org.intermine.bio.web.model.GenomicRegion, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMatchedBaseCount</h4>
<pre>protected&nbsp;java.lang.String&nbsp;getMatchedBaseCount(<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;gr,
                                   java.util.List&lt;java.lang.String&gt;&nbsp;r)</pre>
<div class="block">Calculate the number of matched bases.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>gr</code> - a GenomicRegion object</dd><dd><code>r</code> - a list of attributes</dd>
<dt><span class="strong">Returns:</span></dt><dd>matched base count as String</dd></dl>
</li>
</ul>
<a name="getExportFeaturesQuery(java.util.Set, java.lang.String, java.util.Set, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExportFeaturesQuery</h4>
<pre>public&nbsp;<a href="../../../../../org/intermine/pathquery/PathQuery.html" title="class in org.intermine.pathquery">PathQuery</a>&nbsp;getExportFeaturesQuery(java.util.Set&lt;java.lang.Integer&gt;&nbsp;featureIds,
                               java.lang.String&nbsp;featureType,
                               java.util.Set&lt;java.lang.String&gt;&nbsp;views,
                               java.util.List&lt;java.lang.String&gt;&nbsp;sortOrder)</pre>
<div class="block">A flexiable way of setting query fields.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featureIds</code> - set of feature intermine ids</dd><dd><code>featureType</code> - feature class name</dd><dd><code>views</code> - user defined views in web.properties</dd><dd><code>sortOrder</code> - user defined sortOrder in web.properties</dd>
<dt><span class="strong">Returns:</span></dt><dd>a pathquery</dd></dl>
</li>
</ul>
<a name="getTaxonIds(java.util.Set)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTaxonIds</h4>
<pre>public&nbsp;java.util.Set&lt;java.lang.Integer&gt;&nbsp;getTaxonIds(java.util.Set&lt;java.lang.String&gt;&nbsp;organisms)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>organisms</code> - set of org names</dd>
<dt><span class="strong">Returns:</span></dt><dd>set of taxonIds</dd></dl>
</li>
</ul>
<a name="isJBrowseEnabled()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>isJBrowseEnabled</h4>
<pre>public&nbsp;boolean&nbsp;isJBrowseEnabled()</pre>
<div class="block">Test if jbrowse is enabled</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>boolean</dd></dl>
</li>
</ul>
<a name="getJBrowseTracks()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getJBrowseTracks</h4>
<pre>public&nbsp;java.util.Map&lt;java.lang.String,java.lang.String&gt;&nbsp;getJBrowseTracks()</pre>
<div class="block">Get jbrowse track information from web.properties</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a map: key - feature type, value - track name</dd></dl>
</li>
</ul>
<a name="hasJBrowseTrack(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>hasJBrowseTrack</h4>
<pre>public&nbsp;boolean&nbsp;hasJBrowseTrack(java.lang.String&nbsp;featureType)</pre>
<div class="block">Test if jbrowse has a track regards to a feature type</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featureType</code> - a feature type</dd>
<dt><span class="strong">Returns:</span></dt><dd>boolean</dd></dl>
</li>
</ul>
<a name="generateJBrowseURL(org.intermine.bio.web.model.GenomicRegion, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>generateJBrowseURL</h4>
<pre>public&nbsp;java.lang.String&nbsp;generateJBrowseURL(<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;s,
                                  java.util.List&lt;java.lang.String&gt;&nbsp;featureTypes)</pre>
<div class="block">Generated a jbrowse url</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>s</code> - a GenomicRegion object</dd><dd><code>featureTypes</code> - list of feature types</dd>
<dt><span class="strong">Returns:</span></dt><dd>a string representing jbrowse url</dd></dl>
</li>
</ul>
<a name="generateJBrowseURL(org.intermine.bio.web.model.GenomicRegion)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>generateJBrowseURL</h4>
<pre>public&nbsp;java.lang.String&nbsp;generateJBrowseURL(<a href="../../../../../org/intermine/bio/web/model/GenomicRegion.html" title="class in org.intermine.bio.web.model">GenomicRegion</a>&nbsp;s)</pre>
<div class="block">A wrapper of generateJBrowseURL(GenomicRegion s, List<String> featureTypes)</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>s</code> - GenomicRegion</dd>
<dt><span class="strong">Returns:</span></dt><dd>jbrowse url</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchQueryRunner.html" title="class in org.intermine.bio.web.logic"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/intermine/bio/web/logic/GenomicRegionSearchUtil.html" title="class in org.intermine.bio.web.logic"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/intermine/bio/web/logic/GenomicRegionSearchService.html" target="_top">Frames</a></li>
<li><a href="GenomicRegionSearchService.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
