<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_76) on Wed Apr 06 14:53:18 UTC 2016 -->
<title>FlyBaseProcessor (intermine-javadoc)</title>
<meta name="date" content="2016-04-06">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FlyBaseProcessor (intermine-javadoc)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../org/intermine/bio/dataconversion/FlyBaseIdResolverFactory.html" title="class in org.intermine.bio.dataconversion"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../org/intermine/bio/dataconversion/FlyBaseUTRFastaLoaderTask.html" title="class in org.intermine.bio.dataconversion"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/intermine/bio/dataconversion/FlyBaseProcessor.html" target="_top">Frames</a></li>
<li><a href="FlyBaseProcessor.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.intermine.bio.dataconversion</div>
<h2 title="Class FlyBaseProcessor" class="title">Class FlyBaseProcessor</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li><a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html" title="class in org.intermine.bio.dataconversion">org.intermine.bio.dataconversion.ChadoProcessor</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">org.intermine.bio.dataconversion.SequenceProcessor</a></li>
<li>
<ul class="inheritance">
<li>org.intermine.bio.dataconversion.FlyBaseProcessor</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">FlyBaseProcessor</span>
extends <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></pre>
<div class="block">A converter for chado that handles FlyBase specific configuration.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#FLYBASE_MISCELLANEOUS_CV">FLYBASE_MISCELLANEOUS_CV</a></strong></code>
<div class="block">The cv.name for the FlyBase miscellaneous CV.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#FLYBASE_SO_CV_NAME">FLYBASE_SO_CV_NAME</a></strong></code>
<div class="block">The cv.name for the FlyBase miscellaneous CV.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_org.intermine.bio.dataconversion.SequenceProcessor">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;org.intermine.bio.dataconversion.<a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></h3>
<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#CREATE_SYNONYM_ACTION">CREATE_SYNONYM_ACTION</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#DO_NOTHING_ACTION">DO_NOTHING_ACTION</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#featureMap">featureMap</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#FlyBaseProcessor(org.intermine.bio.dataconversion.ChadoDBConverter)">FlyBaseProcessor</a></strong>(<a href="../../../../org/intermine/bio/dataconversion/ChadoDBConverter.html" title="class in org.intermine.bio.dataconversion">ChadoDBConverter</a>&nbsp;chadoDBConverter)</code>
<div class="block">Create a new FlyBaseChadoDBConverter.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#addCVTermColon(java.lang.String)">addCVTermColon</a></strong>(java.lang.String&nbsp;identifier)</code>
<div class="block">For a FlyBase cvterm identifier like "FBbt00000001", add a colon in the middle and return:
 "FBbt:00000001"</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#createAllelesTempTable(java.sql.Connection)">createAllelesTempTable</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Create a temporary table of allele feature_ids.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#createInsertionTempTable(java.sql.Connection)">createInsertionTempTable</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Create a temporary table from pairs of insertions (eg.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#createLocatedGenesTempTable(java.sql.Connection)">createLocatedGenesTempTable</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Create a temporary table containing the ids of the located genes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#createSynonym(org.intermine.bio.dataconversion.FeatureData,%20java.lang.String)">createSynonym</a></strong>(org.intermine.bio.dataconversion.FeatureData&nbsp;fdat,
             java.lang.String&nbsp;identifier)</code>
<div class="block">Call DataConverter.createSynonym(), store the Item then record in FeatureData that we've
 created it.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#extraProcessing(java.sql.Connection,%20java.util.Map)">extraProcessing</a></strong>(java.sql.Connection&nbsp;connection,
               java.util.Map&lt;java.lang.Integer,org.intermine.bio.dataconversion.FeatureData&gt;&nbsp;features)</code>
<div class="block">For objects that have primaryIdentifier == null, set the primaryIdentifier to be the
 uniquename column from chado.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#fixIdentifier(org.intermine.bio.dataconversion.FeatureData,%20java.lang.String)">fixIdentifier</a></strong>(org.intermine.bio.dataconversion.FeatureData&nbsp;fdat,
             java.lang.String&nbsp;identifier)</code>
<div class="block">Convert ISO entities from FlyBase to HTML entities.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getAlleleCVTermsResultSet(java.sql.Connection)">getAlleleCVTermsResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Get result set of feature_id, cvterm_id pairs for the alleles in flybase chado.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getAllelePropPubResultSet(java.sql.Connection)">getAllelePropPubResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing the featureprop_id and the publication identifier of the
 featureprops for al alleles.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getAllelePropResultSet(java.sql.Connection)">getAllelePropResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing the alleles and their featureprops.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getCDNALengthResultSet(java.sql.Connection)">getCDNALengthResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing the feature_id and its seqlen
 The method is protected so that is can be overridden for
 testing.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getChromosomeStructureVariationResultSet(java.sql.Connection)">getChromosomeStructureVariationResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return the results of running a query for the chromosome_structure_variation feature types.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.util.Map&lt;org.apache.commons.collections.keyvalue.MultiKey,java.util.List&lt;<a href="../../../../org/intermine/bio/chado/config/ConfigAction.html" title="class in org.intermine.bio.chado.config">ConfigAction</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getConfig(int)">getConfig</a></strong>(int&nbsp;taxonId)</code>
<div class="block">Return the config Map.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getDeletionLocationResultSet(java.sql.Connection)">getDeletionLocationResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing location for deletions (chromosome_structure_variation)
 objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getExtraFeatureConstraint()">getExtraFeatureConstraint</a></strong>()</code>
<div class="block">Return an extra constraint to be used when querying the feature table.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.util.List&lt;java.lang.String&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getFeatures()">getFeatures</a></strong>()</code>
<div class="block">Get a list of the chado/so types of the LocatedSequenceFeatures we wish to load.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/intermine/bio/chado/ChadoCV.html" title="class in org.intermine.bio.chado">ChadoCV</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getFlyBaseMiscCV(java.sql.Connection)">getFlyBaseMiscCV</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Get ChadoCV object representing the FlyBase misc cv.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../org/intermine/bio/chado/ChadoCV.html" title="class in org.intermine.bio.chado">ChadoCV</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getFlyBaseSequenceOntologyCV(java.sql.Connection)">getFlyBaseSequenceOntologyCV</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Get ChadoCV object representing SO from FlyBase.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getIndelResultSet(java.sql.Connection)">getIndelResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing pairs of chromosome_structure_variation (deletions) and
 transposable_element_insertion_site (insertions).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getInsertionLocationsResultSet(java.sql.Connection)">getInsertionLocationsResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing pairs of insertion feature_ids (eg. for "FBti0027974" =>
 "FBti0023081") and the fmin and fmax of the first insertion in the pair (ie. the progenitor).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getInteractionResultSet(java.sql.Connection)">getInteractionResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return a result set containing the interaction genes pairs, the title of the publication
 that reported the interaction and its pubmed id.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.sql.ResultSet</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#getLocatedGenesResultSet(java.sql.Connection)">getLocatedGenesResultSet</a></strong>(java.sql.Connection&nbsp;connection)</code>
<div class="block">Return from chado the feature_ids of the genes with entries in the featureloc table.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#makeAnatomyTerm(java.lang.String)">makeAnatomyTerm</a></strong>(java.lang.String&nbsp;identifier)</code>
<div class="block">phenotype annotation creates and stores anatomy terms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#makeFeature(java.lang.Integer,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20int,%20int)">makeFeature</a></strong>(java.lang.Integer&nbsp;featureId,
           java.lang.String&nbsp;chadoFeatureType,
           java.lang.String&nbsp;interMineType,
           java.lang.String&nbsp;name,
           java.lang.String&nbsp;uniqueName,
           int&nbsp;seqlen,
           int&nbsp;taxonId)</code>
<div class="block">Make a new feature</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected org.intermine.bio.dataconversion.FeatureData</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#makeFeatureData(int,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20int,%20int)">makeFeatureData</a></strong>(int&nbsp;featureId,
               java.lang.String&nbsp;type,
               java.lang.String&nbsp;uniqueName,
               java.lang.String&nbsp;name,
               java.lang.String&nbsp;md5checksum,
               int&nbsp;seqlen,
               int&nbsp;organismId)</code>
<div class="block">Create and store a new Item, returning a FeatureData object for the feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.lang.String</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#makeInteractionExperiment(java.lang.String,%20java.lang.String)">makeInteractionExperiment</a></strong>(java.lang.String&nbsp;experimentTitle,
                         java.lang.String&nbsp;publicationItemIdentifier)</code>
<div class="block">Return the item identifier of the Interaction Item for the given pubmed id, creating the
 Item if necessary.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#makeLocation(int,%20int,%20int,%20org.intermine.bio.dataconversion.FeatureData,%20org.intermine.bio.dataconversion.FeatureData,%20int,%20int)">makeLocation</a></strong>(int&nbsp;start,
            int&nbsp;end,
            int&nbsp;strand,
            org.intermine.bio.dataconversion.FeatureData&nbsp;srcFeatureData,
            org.intermine.bio.dataconversion.FeatureData&nbsp;featureData,
            int&nbsp;taxonId,
            int&nbsp;featureId)</code>
<div class="block">note: featureId is needed only by modMine
 Make a Location between a SequenceFeature and a Chromosome.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected java.lang.Integer</code></td>
<td class="colLast"><code><strong><a href="../../../../org/intermine/bio/dataconversion/FlyBaseProcessor.html#store(org.intermine.xml.full.Item,%20int)">store</a></strong>(<a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a>&nbsp;feature,
     int&nbsp;taxonId)</code>
<div class="block">Store the feature Item.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.intermine.bio.dataconversion.SequenceProcessor">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.intermine.bio.dataconversion.<a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></h3>
<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#addToFeatureMap(java.lang.Integer,%20org.intermine.bio.dataconversion.FeatureData)">addToFeatureMap</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#createFeatureTempTable(java.sql.Connection)">createFeatureTempTable</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#earlyExtraProcessing(java.sql.Connection)">earlyExtraProcessing</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#finishedProcessing(java.sql.Connection,%20java.util.Map)">finishedProcessing</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#fixFeatureType(java.lang.String)">fixFeatureType</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#fixIdentifier(org.intermine.bio.dataconversion.FeatureData,%20java.lang.String,%20java.lang.String)">fixIdentifier</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#fixPubMedId(java.lang.String)">fixPubMedId</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getChromosomeFeatureMap(java.lang.Integer)">getChromosomeFeatureMap</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getChromosomeFeatureTypes()">getChromosomeFeatureTypes</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getDbxrefResultSet(java.sql.Connection)">getDbxrefResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatureCVTermResultSet(java.sql.Connection)">getFeatureCVTermResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatureIdQuery()">getFeatureIdQuery</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatureLocResultSet(java.sql.Connection)">getFeatureLocResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatureMap()">getFeatureMap</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeaturePropResultSet(java.sql.Connection)">getFeaturePropResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatureRelationshipResultSet(java.sql.Connection,%20boolean)">getFeatureRelationshipResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatureTableResultSet(java.sql.Connection)">getFeatureTableResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getLibraryCVTermResultSet(java.sql.Connection)">getLibraryCVTermResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getLibraryFeatureResultSet(java.sql.Connection)">getLibraryFeatureResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getMatchLocResultSet(java.sql.Connection)">getMatchLocResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getOrganismConstraint()">getOrganismConstraint</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getPubResultSet(java.sql.Connection)">getPubResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getSegmentedFeatures()">getSegmentedFeatures</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getSynonymResultSet(java.sql.Connection)">getSynonymResultSet</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#initialiseFeatureMap(java.util.Map)">initialiseFeatureMap</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#makePublication(java.lang.Integer)">makePublication</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#process(java.sql.Connection)">process</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#processLocationTable(java.sql.Connection,%20java.sql.ResultSet)">processLocationTable</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#setGeneSource(org.intermine.bio.dataconversion.FeatureData,%20java.lang.String)">setGeneSource</a>, <a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#setGeneSource(java.lang.Integer,%20java.lang.String)">setGeneSource</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.intermine.bio.dataconversion.ChadoProcessor">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.intermine.bio.dataconversion.<a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html" title="class in org.intermine.bio.dataconversion">ChadoProcessor</a></h3>
<code><a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html#getChadoDBConverter()">getChadoDBConverter</a>, <a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html#getDatabase()">getDatabase</a>, <a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html#getItemWriter()">getItemWriter</a>, <a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html#getModel()">getModel</a>, <a href="../../../../org/intermine/bio/dataconversion/ChadoProcessor.html#setAttribute(java.lang.Integer,%20java.lang.String,%20java.lang.String)">setAttribute</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="FLYBASE_MISCELLANEOUS_CV">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FLYBASE_MISCELLANEOUS_CV</h4>
<pre>protected static final&nbsp;java.lang.String FLYBASE_MISCELLANEOUS_CV</pre>
<div class="block">The cv.name for the FlyBase miscellaneous CV.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#org.intermine.bio.dataconversion.FlyBaseProcessor.FLYBASE_MISCELLANEOUS_CV">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="FLYBASE_SO_CV_NAME">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FLYBASE_SO_CV_NAME</h4>
<pre>protected static final&nbsp;java.lang.String FLYBASE_SO_CV_NAME</pre>
<div class="block">The cv.name for the FlyBase miscellaneous CV.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../constant-values.html#org.intermine.bio.dataconversion.FlyBaseProcessor.FLYBASE_SO_CV_NAME">Constant Field Values</a></dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="FlyBaseProcessor(org.intermine.bio.dataconversion.ChadoDBConverter)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FlyBaseProcessor</h4>
<pre>public&nbsp;FlyBaseProcessor(<a href="../../../../org/intermine/bio/dataconversion/ChadoDBConverter.html" title="class in org.intermine.bio.dataconversion">ChadoDBConverter</a>&nbsp;chadoDBConverter)</pre>
<div class="block">Create a new FlyBaseChadoDBConverter.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>chadoDBConverter</code> - the converter that created this object</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getChromosomeStructureVariationResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeStructureVariationResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getChromosomeStructureVariationResultSet(java.sql.Connection&nbsp;connection)
                                                               throws java.sql.SQLException</pre>
<div class="block">Return the results of running a query for the chromosome_structure_variation feature types.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the results</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="createLocatedGenesTempTable(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createLocatedGenesTempTable</h4>
<pre>protected&nbsp;void&nbsp;createLocatedGenesTempTable(java.sql.Connection&nbsp;connection)
                                    throws java.sql.SQLException</pre>
<div class="block">Create a temporary table containing the ids of the located genes.  This is a protected
 method so that it can be overridden for testing</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="createAllelesTempTable(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createAllelesTempTable</h4>
<pre>protected&nbsp;void&nbsp;createAllelesTempTable(java.sql.Connection&nbsp;connection)
                               throws java.sql.SQLException</pre>
<div class="block">Create a temporary table of allele feature_ids.  The table will only have allele of genes
 with locations.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the connection</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="createInsertionTempTable(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createInsertionTempTable</h4>
<pre>protected&nbsp;void&nbsp;createInsertionTempTable(java.sql.Connection&nbsp;connection)
                                 throws java.sql.SQLException</pre>
<div class="block">Create a temporary table from pairs of insertions (eg. "FBti0027974" => "FBti0023081")
 containing the feature_ids of the  pair (the object_id, subject_id in the relation table)
 and the fmin and fmax of the first insertion in the pair (ie. the progenitor / object from
 the feature_relationship table).
 The second in the pair is the "Modified descendant of" the first.  The pairs are found using
 the 'modified_descendant_of' relation type.  All insertions are from DrosDel.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the connection</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getFlyBaseMiscCV(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFlyBaseMiscCV</h4>
<pre>protected&nbsp;<a href="../../../../org/intermine/bio/chado/ChadoCV.html" title="class in org.intermine.bio.chado">ChadoCV</a>&nbsp;getFlyBaseMiscCV(java.sql.Connection&nbsp;connection)
                            throws java.sql.SQLException</pre>
<div class="block">Get ChadoCV object representing the FlyBase misc cv.
 This is a protected method so that it can be overriden for testing</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the database Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the cv</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getFlyBaseSequenceOntologyCV(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFlyBaseSequenceOntologyCV</h4>
<pre>protected&nbsp;<a href="../../../../org/intermine/bio/chado/ChadoCV.html" title="class in org.intermine.bio.chado">ChadoCV</a>&nbsp;getFlyBaseSequenceOntologyCV(java.sql.Connection&nbsp;connection)
                                        throws java.sql.SQLException</pre>
<div class="block">Get ChadoCV object representing SO from FlyBase.
 This is a protected method so that it can be overriden for testing</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the database Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the cv</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="store(org.intermine.xml.full.Item, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>store</h4>
<pre>protected&nbsp;java.lang.Integer&nbsp;store(<a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a>&nbsp;feature,
                      int&nbsp;taxonId)
                           throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></pre>
<div class="block">Store the feature Item.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#store(org.intermine.xml.full.Item,%20int)">store</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>feature</code> - the Item</dd><dd><code>taxonId</code> - the taxon id of this feature</dd>
<dt><span class="strong">Returns:</span></dt><dd>the database id of the new Item</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if an error occurs while storing</dd></dl>
</li>
</ul>
<a name="makeLocation(int, int, int, org.intermine.bio.dataconversion.FeatureData, org.intermine.bio.dataconversion.FeatureData, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>makeLocation</h4>
<pre>protected&nbsp;<a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a>&nbsp;makeLocation(int&nbsp;start,
                int&nbsp;end,
                int&nbsp;strand,
                org.intermine.bio.dataconversion.FeatureData&nbsp;srcFeatureData,
                org.intermine.bio.dataconversion.FeatureData&nbsp;featureData,
                int&nbsp;taxonId,
                int&nbsp;featureId)
                     throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></pre>
<div class="block">note: featureId is needed only by modMine
 Make a Location between a SequenceFeature and a Chromosome.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#makeLocation(int,%20int,%20int,%20org.intermine.bio.dataconversion.FeatureData,%20org.intermine.bio.dataconversion.FeatureData,%20int,%20int)">makeLocation</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>start</code> - the start position</dd><dd><code>end</code> - the end position</dd><dd><code>strand</code> - the strand</dd><dd><code>srcFeatureData</code> - the FeatureData for the src feature (the Chromosome)</dd><dd><code>featureData</code> - the FeatureData for the SequenceFeature</dd><dd><code>taxonId</code> - the taxon id to use when finding the Chromosome for the Location</dd><dd><code>featureId</code> - id of feature</dd>
<dt><span class="strong">Returns:</span></dt><dd>the new Location object</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if there is a problem while storing</dd></dl>
</li>
</ul>
<a name="createSynonym(org.intermine.bio.dataconversion.FeatureData, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createSynonym</h4>
<pre>protected&nbsp;<a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a>&nbsp;createSynonym(org.intermine.bio.dataconversion.FeatureData&nbsp;fdat,
                 java.lang.String&nbsp;identifier)
                      throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></pre>
<div class="block">Call DataConverter.createSynonym(), store the Item then record in FeatureData that we've
 created it.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#createSynonym(org.intermine.bio.dataconversion.FeatureData,%20java.lang.String)">createSynonym</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>fdat</code> - the FeatureData</dd><dd><code>identifier</code> - the identifier to store in the Synonym</dd>
<dt><span class="strong">Returns:</span></dt><dd>the new Synonym</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if there is a problem while storing</dd></dl>
</li>
</ul>
<a name="getLocatedGenesResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLocatedGenesResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getLocatedGenesResultSet(java.sql.Connection&nbsp;connection)
                                               throws java.sql.SQLException</pre>
<div class="block">Return from chado the feature_ids of the genes with entries in the featureloc table.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the db connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the SQL result set</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if a database problem occurs</dd></dl>
</li>
</ul>
<a name="getConfig(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getConfig</h4>
<pre>protected&nbsp;java.util.Map&lt;org.apache.commons.collections.keyvalue.MultiKey,java.util.List&lt;<a href="../../../../org/intermine/bio/chado/config/ConfigAction.html" title="class in org.intermine.bio.chado.config">ConfigAction</a>&gt;&gt;&nbsp;getConfig(int&nbsp;taxonId)</pre>
<div class="block">Return the config Map.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getConfig(int)">getConfig</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>taxonId</code> - return the configuration for this organism</dd>
<dt><span class="strong">Returns:</span></dt><dd>the Map from configuration key to a list of actions</dd></dl>
</li>
</ul>
<a name="getExtraFeatureConstraint()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExtraFeatureConstraint</h4>
<pre>protected&nbsp;java.lang.String&nbsp;getExtraFeatureConstraint()</pre>
<div class="block">Return an extra constraint to be used when querying the feature table.  Any feature table
 column or cvterm table column can be constrained.  The cvterm will match the type_id field
 in the feature.
 eg. "uniquename not like 'BAD_ID%'"</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getExtraFeatureConstraint()">getExtraFeatureConstraint</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the constraint as SQL or nul if there is no extra constraint.</dd></dl>
</li>
</ul>
<a name="makeFeature(java.lang.Integer, java.lang.String, java.lang.String, java.lang.String, java.lang.String, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>makeFeature</h4>
<pre>protected&nbsp;<a href="../../../../org/intermine/xml/full/Item.html" title="class in org.intermine.xml.full">Item</a>&nbsp;makeFeature(java.lang.Integer&nbsp;featureId,
               java.lang.String&nbsp;chadoFeatureType,
               java.lang.String&nbsp;interMineType,
               java.lang.String&nbsp;name,
               java.lang.String&nbsp;uniqueName,
               int&nbsp;seqlen,
               int&nbsp;taxonId)</pre>
<div class="block">Make a new feature</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#makeFeature(java.lang.Integer,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20int,%20int)">makeFeature</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>featureId</code> - the chado feature id</dd><dd><code>chadoFeatureType</code> - the chado feature type (a SO term)</dd><dd><code>interMineType</code> - the InterMine type of the feature</dd><dd><code>name</code> - the name</dd><dd><code>uniqueName</code> - the uniquename</dd><dd><code>seqlen</code> - the sequence length (if known)</dd><dd><code>taxonId</code> - the NCBI taxon id of the current feature</dd>
<dt><span class="strong">Returns:</span></dt><dd>the new feature</dd></dl>
</li>
</ul>
<a name="getFeatures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre>protected&nbsp;java.util.List&lt;java.lang.String&gt;&nbsp;getFeatures()</pre>
<div class="block">Get a list of the chado/so types of the LocatedSequenceFeatures we wish to load.  The list
 will not include chromosome-like features.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#getFeatures()">getFeatures</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the list of features</dd></dl>
</li>
</ul>
<a name="extraProcessing(java.sql.Connection, java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>extraProcessing</h4>
<pre>protected&nbsp;void&nbsp;extraProcessing(java.sql.Connection&nbsp;connection,
                   java.util.Map&lt;java.lang.Integer,org.intermine.bio.dataconversion.FeatureData&gt;&nbsp;features)
                        throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a>,
                               java.sql.SQLException</pre>
<div class="block">For objects that have primaryIdentifier == null, set the primaryIdentifier to be the
 uniquename column from chado.
 Do any extra processing for this database, after all other processing is done</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#extraProcessing(java.sql.Connection,%20java.util.Map)">extraProcessing</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd><dd><code>features</code> - a map from chado feature_id to data for that feature</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if there is a problem while storing</dd>
<dd><code>java.sql.SQLException</code> - if there is a problem</dd></dl>
</li>
</ul>
<a name="makeInteractionExperiment(java.lang.String, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>makeInteractionExperiment</h4>
<pre>protected&nbsp;java.lang.String&nbsp;makeInteractionExperiment(java.lang.String&nbsp;experimentTitle,
                                         java.lang.String&nbsp;publicationItemIdentifier)
                                              throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></pre>
<div class="block">Return the item identifier of the Interaction Item for the given pubmed id, creating the
 Item if necessary.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>experimentTitle</code> - the new title</dd><dd><code>publicationItemIdentifier</code> - the item identifier of the publication for this experiment</dd>
<dt><span class="strong">Returns:</span></dt><dd>the interaction item identifier</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if the item can't be stored</dd></dl>
</li>
</ul>
<a name="getAlleleCVTermsResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlleleCVTermsResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getAlleleCVTermsResultSet(java.sql.Connection&nbsp;connection)
                                                throws java.sql.SQLException</pre>
<div class="block">Get result set of feature_id, cvterm_id pairs for the alleles in flybase chado.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connectio</dd>
<dt><span class="strong">Returns:</span></dt><dd>the cvterms</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="addCVTermColon(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addCVTermColon</h4>
<pre>protected static&nbsp;java.lang.String&nbsp;addCVTermColon(java.lang.String&nbsp;identifier)</pre>
<div class="block">For a FlyBase cvterm identifier like "FBbt00000001", add a colon in the middle and return:
 "FBbt:00000001"</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>identifier</code> - the identifier from chado</dd>
<dt><span class="strong">Returns:</span></dt><dd>the public identifier</dd></dl>
</li>
</ul>
<a name="makeAnatomyTerm(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>makeAnatomyTerm</h4>
<pre>protected&nbsp;java.lang.String&nbsp;makeAnatomyTerm(java.lang.String&nbsp;identifier)
                                    throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></pre>
<div class="block">phenotype annotation creates and stores anatomy terms.  so does librarycvterm</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#makeAnatomyTerm(java.lang.String)">makeAnatomyTerm</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>identifier</code> - identifier for anatomy term</dd>
<dt><span class="strong">Returns:</span></dt><dd>refId for anatomy term object</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if term can't be stored</dd></dl>
</li>
</ul>
<a name="getInteractionResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInteractionResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getInteractionResultSet(java.sql.Connection&nbsp;connection)
                                              throws java.sql.SQLException</pre>
<div class="block">Return a result set containing the interaction genes pairs, the title of the publication
 that reported the interaction and its pubmed id.  The method is protected
 so that is can be overridden for testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getAllelePropResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllelePropResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getAllelePropResultSet(java.sql.Connection&nbsp;connection)
                                             throws java.sql.SQLException</pre>
<div class="block">Return a result set containing the alleles and their featureprops.  The method is protected
 so that is can be overridden for testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getIndelResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIndelResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getIndelResultSet(java.sql.Connection&nbsp;connection)
                                        throws java.sql.SQLException</pre>
<div class="block">Return a result set containing pairs of chromosome_structure_variation (deletions) and
 transposable_element_insertion_site (insertions).  The method is protected
 so that is can be overridden for testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getInsertionLocationsResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInsertionLocationsResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getInsertionLocationsResultSet(java.sql.Connection&nbsp;connection)
                                                     throws java.sql.SQLException</pre>
<div class="block">Return a result set containing pairs of insertion feature_ids (eg. for "FBti0027974" =>
 "FBti0023081") and the fmin and fmax of the first insertion in the pair (ie. the progenitor).
 The second in the pair is the "Modified descendant of" the first.  The pairs are found using
 the 'modified_descendant_of' relation type.  All insertions are from DrosDel.
 The method is protected so that is can be overridden for testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getDeletionLocationResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDeletionLocationResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getDeletionLocationResultSet(java.sql.Connection&nbsp;connection)
                                                   throws java.sql.SQLException</pre>
<div class="block">Return a result set containing location for deletions (chromosome_structure_variation)
 objects.  The locations are in the featureprop able in the form:
   2R:12716549..12984803 (53D11;53F8)
 The method is protected so that is can be overridden for testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getAllelePropPubResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllelePropPubResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getAllelePropPubResultSet(java.sql.Connection&nbsp;connection)
                                                throws java.sql.SQLException</pre>
<div class="block">Return a result set containing the featureprop_id and the publication identifier of the
 featureprops for al alleles.  The method is protected so that is can be overridden for
 testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="getCDNALengthResultSet(java.sql.Connection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDNALengthResultSet</h4>
<pre>protected&nbsp;java.sql.ResultSet&nbsp;getCDNALengthResultSet(java.sql.Connection&nbsp;connection)
                                             throws java.sql.SQLException</pre>
<div class="block">Return a result set containing the feature_id and its seqlen
 The method is protected so that is can be overridden for
 testing.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>connection</code> - the Connection</dd>
<dt><span class="strong">Returns:</span></dt><dd>the ResultSet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code>java.sql.SQLException</code> - if there is a database problem</dd></dl>
</li>
</ul>
<a name="fixIdentifier(org.intermine.bio.dataconversion.FeatureData, java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fixIdentifier</h4>
<pre>protected&nbsp;java.lang.String&nbsp;fixIdentifier(org.intermine.bio.dataconversion.FeatureData&nbsp;fdat,
                             java.lang.String&nbsp;identifier)</pre>
<div class="block">Convert ISO entities from FlyBase to HTML entities.
 Process the identifier and return a "cleaned" version.  Implement in sub-classes to fix
 data problem.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#fixIdentifier(org.intermine.bio.dataconversion.FeatureData,%20java.lang.String)">fixIdentifier</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>fdat</code> - the FeatureData object of the feature that this identifier came from</dd><dd><code>identifier</code> - the identifier</dd>
<dt><span class="strong">Returns:</span></dt><dd>a cleaned identifier</dd></dl>
</li>
</ul>
<a name="makeFeatureData(int, java.lang.String, java.lang.String, java.lang.String, java.lang.String, int, int)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>makeFeatureData</h4>
<pre>protected&nbsp;org.intermine.bio.dataconversion.FeatureData&nbsp;makeFeatureData(int&nbsp;featureId,
                                                           java.lang.String&nbsp;type,
                                                           java.lang.String&nbsp;uniqueName,
                                                           java.lang.String&nbsp;name,
                                                           java.lang.String&nbsp;md5checksum,
                                                           int&nbsp;seqlen,
                                                           int&nbsp;organismId)
                                                                throws <a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></pre>
<div class="block">Create and store a new Item, returning a FeatureData object for the feature.</div>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html#makeFeatureData(int,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20int,%20int)">makeFeatureData</a></code>&nbsp;in class&nbsp;<code><a href="../../../../org/intermine/bio/dataconversion/SequenceProcessor.html" title="class in org.intermine.bio.dataconversion">SequenceProcessor</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>featureId</code> - the chado id from the feature table</dd><dd><code>type</code> - the type of the feature from the feature + cvterm tables</dd><dd><code>uniqueName</code> - the uniquename from chado</dd><dd><code>name</code> - the name from chado</dd><dd><code>md5checksum</code> - the checksum from the chado feature able</dd><dd><code>seqlen</code> - the length from the feature table</dd><dd><code>organismId</code> - the organism id of the feature from chado</dd>
<dt><span class="strong">Returns:</span></dt><dd>a FeatureData object</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/intermine/objectstore/ObjectStoreException.html" title="class in org.intermine.objectstore">ObjectStoreException</a></code> - if there is a problem while storing</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../org/intermine/bio/dataconversion/FlyBaseIdResolverFactory.html" title="class in org.intermine.bio.dataconversion"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../org/intermine/bio/dataconversion/FlyBaseUTRFastaLoaderTask.html" title="class in org.intermine.bio.dataconversion"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/intermine/bio/dataconversion/FlyBaseProcessor.html" target="_top">Frames</a></li>
<li><a href="FlyBaseProcessor.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
